From 238ac06ed915cde70d9a5f0ef3c4ec97cb9f47c9 Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Fri, 2 Oct 2015 10:41:37 -0700
Subject: [PATCH 14/17] input: silead: add irq enable/disable in suspend/resume

Disabled IRQ during suspend and enabled it during resume.

Also added power off in resume code to handle a hardware
reset issue.

Change-Id: I89d9451217c508b209b86d1370c4521844fe16e9
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Reviewed-on: https://android.intel.com:443/472787
---
 drivers/input/touchscreen/silead.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 94dd3fb..c4b225a 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -521,6 +521,7 @@ static int silead_ts_suspend(struct device *dev)
 {
 	struct i2c_client *client = to_i2c_client(dev);
 
+	disable_irq(client->irq);
 	silead_ts_set_power(client, SILEAD_POWER_OFF);
 	msleep(20);
 	return 0;
@@ -531,6 +532,9 @@ static int silead_ts_resume(struct device *dev)
 	struct i2c_client *client = to_i2c_client(dev);
 	int ret, status;
 
+	enable_irq(client->irq);
+	/* send power off again, to handle some hardware reset issue */
+	silead_ts_set_power(client, SILEAD_POWER_OFF);
 	silead_ts_set_power(client, SILEAD_POWER_ON);
 	msleep(20);
 
-- 
1.9.1

